packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantAvailability/RestaurantAvailability.vue

```diff
@@ -0,0 +1,106 @@
+        <icon-text
+            v-if="availabilityTranslatedName"
+            :text="availabilityTranslatedName"
+            :color="iconColor"
+            hide-icon-in-tile-view
+            data-test-id="restaurant-availability-type"
+            :class="[$style['c-restaurantCard-availability-iconText']]">
+            <component
+                :is="icon"
+                v-if="icon"
+                aria-hidden="true"
+                data-test-id="restaurant-availability-icon" />
+        </icon-text>
+
+        <p
+            v-if="detailsMessage"
+            data-test-id="restaurant-availability-detailsMessage"
+            :class="[$style['c-restaurantCard-availability-detailsMessage']]">
+            {{ detailsMessage }}
+        </p>
+    </div>
+</template>
+
+<script>
+import { ClockSmallIcon, WalkingSmallIcon } from '@justeat/f-vue-icons';
+import isValidAvailabilityType from './propValidators/availabilityType.validator';
+import availabilityTypes from './availabilityTypes';
+import IconText from '../IconText.vue';
+
+const availabilityTypeIcons = {
+    [availabilityTypes.PREORDER]: ClockSmallIcon,
+    [availabilityTypes.COLLECTION]: WalkingSmallIcon
+};
+
+const iconColors = {
+    [availabilityTypes.PREORDER]: 'colorSupportInfo'
+};
+
+export default {
+    name: 'RestaurantAvailability',
+    components: {
+        IconText,
+        ClockSmallIcon,
+        WalkingSmallIcon
+    },
+    props: {
+        // A string representation of pre-defined availability types: PREORDER, COLLECTION
+        availabilityType: {
+            type: String,
+            default: null,
+            validator: isValidAvailabilityType
+        },
+        // The translated version of the availability type for the current tenant i.e. 'Pre-order', 'Preordina'
+        availabilityTranslatedName: {
+            type: String,
+            default: null
+        },
+        // The accompanying message for the availability type. i.e Opening times, delivering from...
+        detailsMessage: {
+            type: String,
+            default: null
+        }
+    },
+    computed: {
+        icon () {
+            return availabilityTypeIcons[this.availabilityType] || null;
+        },
+        iconColor () {
+            return iconColors[this.availabilityType] || null;
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+.c-restaurantCard-availability {
+    display: flex;
+    align-items: center;
+    flex-wrap: wrap;
+
+    svg {
+        height: 16px;
+    }
+}
+
+.c-restaurantCard-availability-iconText,
+.c-restaurantCard-availability-detailsMessage {
+    overflow-wrap: break-word;
+    @include font-size($font-paragraph-02);
+}
+
+.c-restaurantCard-availability-iconText {
+    &:after {
+        content: '\2022'; // round bullet character
+        color: $color-content-subdued;
+        line-height: 1;
+        margin: 0 spacing(x0.5);
+    }
+}
+
+.c-restaurantCard-availability-detailsMessage {
+    margin-top: 0;
```

Comment:
where does the margin come from that needs resetting? 
Should  we be using a modifier to apply to those that need vs applying to all then removing?

Reply:
it was from the `<p>` being used for it - I just switched to a `<span>` and removed the margin override :)